crates/astria-account-monitor/src/account_monitor.rs

```diff
@@ -0,0 +1,296 @@
+    sequencer_accounts: SequencerAccountsToMonitor,
+    sequencer_asset: Asset,
+    metrics: &'static Metrics,
+    interval: Duration,
+}
+
+impl AccountMonitor {
+    /// Instantiates a new `Service`.
+    ///
+    /// # Errors
+    ///
+    /// - If the provided `sequencer_abci_endpoint` string cannot be contructed to a cometbft http
+    ///   client.
+    /// - If the provided `sequencer_asset` string cannot be parsed to a valid asset.
+    /// - If the provided `sequencer_accounts` string cannot be parsed to a valid address.
+    /// - If the provided `sequencer_bridge_accounts` string cannot be parsed to a valid address.
+    #[instrument(skip_all, err)]
+    pub fn new(cfg: Config, metrics: &'static Metrics) -> eyre::Result<Self> {
+        let shutdown_handle = ShutdownHandle::new();
+
+        let Config {
+            sequencer_abci_endpoint,
+            sequencer_asset,
+            sequencer_accounts,
+            ..
+        } = cfg;
+
+        let sequencer_cometbft_client =
+            sequencer_client::HttpClient::new(&*sequencer_abci_endpoint).wrap_err_with(|| {
+                format!("failed to create sequencer client for url {sequencer_abci_endpoint}")
+            })?;
+
+        let interval = Duration::from_millis(cfg.query_interval_ms);
+        Ok(Self {
+            shutdown_token: shutdown_handle,
+            sequencer_abci_client: sequencer_cometbft_client,
+            sequencer_accounts,
+            sequencer_asset,
+            metrics,
+            interval,
+        })
+    }
+
+    /// Run the query loop, polling the sequencer for accounts information.
+    ///
+    /// # Errors
+    /// An error is returned if bridge last transaction height is not found.
+    pub async fn run(&self) -> eyre::Result<()> {
+        let Some(_res) = self
+            .shutdown_token
+            .token
+            .run_until_cancelled(run_loop(
+                self.metrics,
+                &self.sequencer_abci_client,
+                &self.sequencer_accounts,
+                &self.sequencer_asset,
+                self.interval,
+            ))
+            .await
+        else {
+            return Ok(());
+        };
+
+        Ok(())
+    }
+}
+
+async fn run_loop(
+    metrics: &'static Metrics,
+    client: &sequencer_client::HttpClient,
+    accounts: &SequencerAccountsToMonitor,
+    asset: &Asset,
+    pull_interval: Duration,
+) {
+    let mut poll_timer = interval(pull_interval);
+    poll_timer.set_missed_tick_behavior(tokio::time::MissedTickBehavior::Skip);
+
+    loop {
+        poll_timer.tick().await;
+
+        fetch_all_info(metrics, client, accounts, asset);
+    }
+}
+
+fn fetch_all_info(
+    metrics: &'static Metrics,
+    client: &sequencer_client::HttpClient,
+    accounts: &SequencerAccountsToMonitor,
+    asset: &Asset,
+) {
+    for account in accounts {
+        let client = client.clone();
+        let asset = asset.clone();
+        let address = account.address;
+        let _handle =
```

Comment:
You don't need to `let _handle = ` I think - you should be able to just `tokio::spawn` without binding to a variable (or did clippy flag this with `must_use`?)